src/core/logger/nest-logger-adapter.service.ts

```diff
@@ -54,8 +54,7 @@ export class NestLoggerAdapterService implements INestLogger, OnModuleInit {
   }
 
   private mapName(context?: string) {
-    return context
-      ? NestLoggerAdapterService.nameMap[context] || context
-      : 'nest';
+    // @ts-ignore FIXME: unclear how to get around the TS error here
+    return context ? NestLoggerAdapterService.nameMap[context] || context : 'nest';
```

Comment:
I believe we just need to be explicit with the type definition of `nameMap` to loosen the keys. 

```tsx
private static nameMap: Record<string, string> = {
  NestFactory: 'nest:factory',
  ...
};
```

Reply:
Perfect, works :)

Reply:
Perfect!